### PR TITLE
Base Prometheus metrics handler

### PR DESCRIPTION
**Pull request by AlekseyLobanov**
I want to add some standard metrics to navidrome. Prometheus metrics are useful for observability (including alerts).
I believe that many simple PRs are much better than 1 full.

What this PR has:
1. Two options for enable/disable (default for compatibility) metrics and their path
2. Standard prometheus handler with default golang metrics like `process_start_time_seconds`, `go_threads` and `process_cpu_seconds_total`
3. Does not consume resources if disabled.

I believe that metrics path is best approach, compare to different port or some auth, including HTTP Basic Auth:
1. If navidrome works behind reverse proxy, closing `/metrics` is the most simple way to protect them
2. Otherwise, metrics path like `/metrics_super_secret_path` is simple to use and secure enough. The most serious problem in this case is plain HTTP connection over not secure connection which is not secure anyway.

This relates to #1548 , but not solves initial request for now

image

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -191,12 +196,19 @@ func init() {
 	rootCmd.Flags().String("imagecachesize", viper.GetString("imagecachesize"), "size of image (art work) cache. set to 0 to disable cache")
 	rootCmd.Flags().Bool("autoimportplaylists", viper.GetBool("autoimportplaylists"), "enable/disable .m3u playlist auto-import`")
 
+	rootCmd.Flags().Bool("prometheus.enabled", viper.GetBool("prometheus.enabled"), "enable/disable prometheus metrics endpoint`")
+	rootCmd.Flags().String("prometheus.metricspath", viper.GetString("prometheus.metricspath"), "size of image (art work) cache. set to 0 to disable cache")
```
Comment (vs49688) [inline cmd/root.go:200]: Forget to change the description?

Comment (AlekseyLobanov) [reply cmd/root.go]: Yes, unfortunately. Fixed

**Comment by deluan** [comment]
Hey, sorry for the long delay! LGTM, just need to rebase with latest from master and I'll merge it right away. Thanks!

**Comment by AlekseyLobanov** [comment]
Branch is now actual

**Comment by deluan** [comment]
Thanks!

**Comment by anarcat** [comment]
hi! are there any plans to expand this to (say) count the number of users, albums, artists, tracks, etc?

**Comment by djjudas21** [comment]
Thanks for the addition. How is this feature enabled with env vars? I need to add support for metrics to the helm chart but I'm not sure how the Go options correspond to env vars. Is `prometheus.enabled` equivalent to `ND_PROMETHEUS_ENABLED`?

**Comment by AlekseyLobanov** [comment]
> How is this feature enabled with env vars? I need to add support for metrics to the helm chart but I'm not sure how the Go options correspond to env vars. Is `prometheus.enabled` equivalent to `ND_PROMETHEUS_ENABLED`?

I should add this to docs repository, definitely.

To enable metrics:
1. Define variable `ND_PROMETHEUS_ENABLED=true`
2. (optional) Use custom metrics path `ND_PROMETHEUS_METRICSPATH=/metrics_secret_key`

**Comment by AlekseyLobanov** [comment]
> hi! are there any plans to expand this to (say) count the number of users, albums, artists, tracks, etc?

Of course!

It was my first PR to this project (and first PR to golang project), I tried to create it as small as possible. It has only required dependencies and configuration layer required by Prometheus to run.

Next step is adding navidrome-specfic metrics. I have plans to do this in near future.

**Pull request merged**